src/main/fc/tasks.c

```diff
@@ -113,11 +113,32 @@
 
 static void taskMain(timeUs_t currentTimeUs)
 {
-    UNUSED(currentTimeUs);
-
 #ifdef USE_SDCARD
     afatfs_poll();
 #endif
+
+#ifdef USE_DSHOT_TELEMETRY
+    static bool dshotTelemetryActive = false;
+    static timeUs_t nextDshoTelemetryCheck = 0;
+
+    if (!ARMING_FLAG(ARMED)) {
+        if (motorConfig()->dev.useDshotTelemetry && !dshotTelemetryActive && (currentTimeUs > nextDshoTelemetryCheck))  {
```

Comment:
I'd reorder this to bring `currentTimeUs > nextDshoTelemetryCheck` to the start of the condition, to allow lazy evaluation to take the shortest way 999 times out of 1000.
Or even check the timing in the outer `if`.

Also, using `cmpTimeUs()` is a better way to compare timestamps.

Reply:
Well I was trying to minimize cycles during critical periods by having the outside condition only be the not armed test.  But I could also move the `!dshotTelemetryActive` to the outside loop to limit processing after the telemetry gets initialized. No problem on switching to the `cmpTimeUs()`.